File: Cpp/557.cpp
#include <iostream>
#include <string>

class Solution {
public:
    string reverseWords(string s) {
        int i = 0;
        int lenth = s.size();
        while(i < lenth) {
            int start = i;
            while(i < lenth && s[i] != ' ') {
                i ++;
            }
            int end = i - 1;
            while(start < end) {
                int temp = s[start];
                s[start] = s[end];
                s[end] = temp;
                end --;
                start ++;
            }
            while(i < lenth && s[i] == ' ') {
                i ++;
            }
        }
        return s;
    }
};

File: go/557.go
package Leetcode

func reverseWords(s string) string {
	res := []byte{}
	lenth := len(s)
	for i := 0; i < lenth; {
		start := i
		for i < lenth && s[i] != ' ' {
			i++
		}
		end := i - 1
		for end >= start {
			res = append(res, s[end])
			end--
		}
		for i < lenth && s[i] == ' ' {
			i++
			res = append(res, ' ')
		}
	}
	return string(res)
}
